server/statistics/store.go

```diff
@@ -79,187 +77,36 @@ func (s *StoresStats) Set(storeID uint64, stats *pdpb.StoreStats) {
 	store.Set(stats)
 }
 
-// UpdateTotalBytesRate updates the total bytes write rate and read rate.
-func (s *StoresStats) UpdateTotalBytesRate(f func() []*core.StoreInfo) {
-	var totalBytesWriteRate float64
-	var totalBytesReadRate float64
-	var writeRate, readRate float64
-	ss := f()
-	s.RLock()
-	defer s.RUnlock()
-	for _, store := range ss {
-		if store.IsUp() {
-			stats, ok := s.rollingStoresStats[store.GetID()]
-			if !ok {
-				continue
-			}
-			writeRate, readRate = stats.GetBytesRate()
-			totalBytesWriteRate += writeRate
-			totalBytesReadRate += readRate
-		}
+// UpdateTotalLoad updates the total loads of all stores.
+func (s *StoresStats) UpdateTotalLoad(stores []*core.StoreInfo) {
+	s.Lock()
+	defer s.Unlock()
+	for i := range s.totalLoads {
+		s.totalLoads[i] = 0
 	}
-	s.bytesWriteRate = totalBytesWriteRate
-	s.bytesReadRate = totalBytesReadRate
-}
-
-// UpdateTotalKeysRate updates the total keys write rate and read rate.
-func (s *StoresStats) UpdateTotalKeysRate(f func() []*core.StoreInfo) {
-	var totalKeysWriteRate float64
-	var totalKeysReadRate float64
-	var writeRate, readRate float64
-	ss := f()
-	s.RLock()
-	defer s.RUnlock()
-	for _, store := range ss {
-		if store.IsUp() {
-			stats, ok := s.rollingStoresStats[store.GetID()]
-			if !ok {
-				continue
-			}
-			writeRate, readRate = stats.GetKeysRate()
-			totalKeysWriteRate += writeRate
-			totalKeysReadRate += readRate
+	for _, store := range stores {
```

Comment:
why to set zero

Reply:
To reuse the map. Don't have to allocate new map every time.